src/modules/clock.cpp

```diff
@@ -11,46 +24,39 @@ waybar::modules::Clock::Clock(const std::string& id, const Json::Value& config)
   };
 }
 
+using zoned_time = date::zoned_time<std::chrono::system_clock::duration>;
+
+struct waybar_time {
+  std::locale locale;
+  zoned_time ztime;
+};
+
 auto waybar::modules::Clock::update() -> void {
-  tzset(); // Update timezone information
-  const date::time_zone* zone;
-  auto now = std::chrono::floor<std::chrono::seconds>(std::chrono::system_clock::now());
-  if (config_["timezone"].isString()) {
-    zone = date::locate_zone(config_["timezone"].asString());
-  } else {
-    zone = date::current_zone();
+  if (!fixed_time_zone_) {
```

Comment:
Why not just use `!config_["timezone"].isString()` and removing `fixed_time_zone_` var?

Reply:
To avoid going trough the json parser each time. May not be worth it.

Reply:
The json is already parsed, but yeah, never mind.